.github/workflows/cypress-test-multiauth-e2e.yml

```diff
@@ -82,14 +83,30 @@ jobs:
           opensearch_security.multitenancy.tenants.preferred: ["Private", "Global"]
           opensearch_security.readonly_mode.roles: ["kibana_read_only"]
           opensearch_security.cookie.secure: false
-          server.xsrf.allowlist: ["/_plugins/_security/api/authtoken", "/_opendistro/_security/api/authtoken", "/_opendistro/_security/saml/acs", "/_opendistro/_security/saml/acs/idpinitiated", "/_opendistro/_security/saml/logout"]
+          server.xsrf.allowlist: ["/_opendistro/_security/saml/acs", "/_opendistro/_security/saml/acs/idpinitiated", "/_opendistro/_security/saml/logout"]
           opensearch_security.auth.type: ["basicauth","saml"]
           opensearch_security.auth.multiple_auth_enabled: true
           opensearch_security.auth.anonymous_auth_enabled: false
           home.disableWelcomeScreen: true
           EOT
 
+      - name: Run OSD with basePath
+        if: ${{ matrix.basePath != '' }}
+        run: |
+          echo "server.basePath: \"${{ matrix.basePath }}\"" >> opensearch_dashboards_multiauth.yml
+          echo "server.rewriteBasePath: true" >> opensearch_dashboards_multiauth.yml
+
+      - name: Run Cypress Tests with basePath
+        if: ${{ matrix.basePath != '' }}
+        uses: ./.github/actions/run-cypress-tests
+        with:
+          security_config_file: config_multiauth.yml
+          dashboards_config_file: opensearch_dashboards_multiauth.yml
+          yarn_command: 'yarn cypress:run --browser chrome --headless --env loginMethod=saml_multiauth,basePath=${{ matrix.basePath }} --spec "test/cypress/e2e/saml/*.js"'
+          osd_base_path: ${{ matrix.basePath }}
+
       - name: Run Cypress Tests
+        if: ${{ matrix.basePath == '' }}
```

Comment:
I think you can probably combine these two/ the run cypress tests action is already simply evaluating the yarn command, so can't you simple pass in basePath even if it is '/osd' or '', instead of separating it out into two blocks?

Reply:
+1. We should be able to combine these, since the default base-path value is empty-string which is equivalent to no base-path

Reply:
The `yarn_command` is different in both of these. I could try using the ternary operator inline to reduce this to a single step, but this was how I was able to get it working.